kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/security/adapters/JMSSecurityAdapter.java

```diff
@@ -105,14 +113,22 @@ private static UserDetails getUserDetails(final String user, final String pass)
         }
     }
 
-    private static UserDetails getElytronUserDetails(final String user, final String password) throws RealmUnavailableException {
+    private static UserDetails getElytronUserDetails(final String user, final String password) {
         final UserDetails userDetails = new UserDetails();
         final List<String> roles = new ArrayList<>();
 
-        final SecurityIdentity identity = SecurityDomain.getCurrent().authenticate(user, new PasswordGuessEvidence(password.toCharArray()));
-        userDetails.setName(identity.getPrincipal().getName());
-        for (String role : identity.getRoles()) {
-            roles.add(role);
+        if (isElytronActive) {
+            try {
+                final SecurityIdentity identity = SecurityDomain.getCurrent().authenticate(user,
+                        (Evidence) passwordGuessEvidenceClass.getDeclaredConstructor(char[].class).newInstance(password.toCharArray()));
```

Comment:
Without this workaround, `kie-server-tests` were still failing complaining about `Evidence` class:
```
[INFO] [talledLocalContainer] java.util.ServiceConfigurationError: org.kie.server.api.security.SecurityAdapter: org.kie.server.services.impl.security.adapters.JMSSecurityAdapter Unable to get public no-arg constructor
 [INFO] [talledLocalContainer]   at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:582)
 [INFO] [talledLocalContainer]   at java.base/java.util.ServiceLoader.getConstructor(ServiceLoader.java:673)
 [INFO] [talledLocalContainer]   at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1233)
 [INFO] [talledLocalContainer]   at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1265)
 [INFO] [talledLocalContainer]   at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1300)
 [INFO] [talledLocalContainer]   at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1385)
...
 [INFO] [talledLocalContainer] Caused by: java.lang.NoClassDefFoundError: org/wildfly/security/evidence/Evidence
 [INFO] [talledLocalContainer]   at java.base/java.lang.Class.getDeclaredConstructors0(Native Method)
 [INFO] [talledLocalContainer]   at java.base/java.lang.Class.privateGetDeclaredConstructors(Class.java:3137)
 [INFO] [talledLocalContainer]   at java.base/java.lang.Class.getConstructor0(Class.java:3342)
 [INFO] [talledLocalContainer]   at java.base/java.lang.Class.getConstructor(Class.java:2151)
 [INFO] [talledLocalContainer]   at java.base/java.util.ServiceLoader$1.run(ServiceLoader.java:660)
 [INFO] [talledLocalContainer]   at java.base/java.util.ServiceLoader$1.run(ServiceLoader.java:657)
 [INFO] [talledLocalContainer]   at java.base/java.security.AccessController.doPrivileged(Native Method)
 [INFO] [talledLocalContainer]   at java.base/java.util.ServiceLoader.getConstructor(ServiceLoader.java:668)
 [INFO] [talledLocalContainer]   ... 52 more
 [INFO] [talledLocalContainer] Caused by: java.lang.ClassNotFoundException: org.wildfly.security.evidence.Evidence
 [INFO] [talledLocalContainer]   at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1365)
 [INFO] [talledLocalContainer]   at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1188)
 [INFO] [talledLocalContainer]   ... 60 more
```

Tests were ran using following command:
```
mvn -f kie-server-parent/kie-server-tests/pom.xml -Dcontainer=tomcat9 -Dcontainer.startstop.timeout.millis=240000 -Deap7.download.url=http://download.devel.redhat.com/released/JBoss-middleware/eap7/7.4.10/jboss-eap-7.4.10.zip -Dmaven.test.failure.ignore=true -Ddeployment.timeout.millis=240000 -B -e -fae -nsu verify -Ptomcat9
```

Reply:
Mmm ok it seems there are still some other parts that I missed, checking..